x-pack/elastic-agent/pkg/agent/cmd/run.go

```diff
@@ -296,3 +314,52 @@ func setupMetrics(agentInfo *info.AgentInfo, logger *logger.Logger, operatingSys
 func isProcessStatsEnabled(cfg *monitoringCfg.MonitoringHTTPConfig) bool {
 	return cfg != nil && cfg.Enabled
 }
+
+func tryDelayEnroll(ctx context.Context, logger *logger.Logger, cfg *configuration.Configuration, override cfgOverrider) (*configuration.Configuration, error) {
+	enrollPath := paths.AgentEnrollFile()
+	if _, err := os.Stat(enrollPath); os.IsNotExist(err) {
+		// no enrollment file exists; nothing to do
+		return cfg, nil
```

Comment:
override is not called here, shouldnt this be called on defer maybe

Reply:
The `cfg` at that point has already had `overrides` performed on it. `overrides` only needs to be used again in the case that enroll did occur and the entire configuration needs to be read again.

That is why `overrides` is only passed into the `loadConfig` function, and should only be done in the case that the config truly needs to be reloaded.

Reply:
i see i havent noticed call before